File: Backend/IBM/IBM/src/main/java/IBM/Colombia/Cia/S/C/A/IBM/application/dto/response/AuthResponse.java
package IBM.Colombia.Cia.S.C.A.IBM.application.dto.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class AuthResponse {
    private TokenInfo token;
    @JsonProperty("user_information")
    private SoapData userInformation;

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class TokenInfo {
        private String token;
        private String type = "Bearer";
    }

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class SoapData {
        private int resultCode;
        private String firstName;
        private String lastName;
        private int age;
        private String profilePhoto;
        private String video;
    }
}

File: Backend/IBM-NODEJS/src/application/dto/response/AuthResponse.ts
export class AuthResponse {
    constructor(
        public token?: TokenInfo,
        public user_information?: SoapData
    ) {}
}

export class TokenInfo {
    constructor(
        public token?: string,
        public type?: string
    ) {}
}

export class SoapData {
    constructor(
        public resultCode?: number,
        public firstName?: string,
        public lastName?: string,
        public age?: number,
        public profilePhoto?: string,
        public video?: string
    ) {}
}
